sklearn/ensemble/_iforest.py

```diff
@@ -477,19 +489,20 @@ def _compute_score_samples(self, X, subsample_features):
 
         depths = np.zeros(n_samples, order="f")
 
-        for tree, features in zip(self.estimators_, self.estimators_features_):
+        for tree_idx, (tree, features) in enumerate(
+            zip(self.estimators_, self.estimators_features_)
+        ):
             X_subset = X[:, features] if subsample_features else X
 
             leaves_index = tree.apply(X_subset)
-            node_indicator = tree.decision_path(X_subset)
-            n_samples_leaf = tree.tree_.n_node_samples[leaves_index]
+            path_lengths = self._decision_path_lengths[tree_idx][leaves_index]
```

Comment:
we can remove this line and directly put the expression in the `depths` computation